distributed/diagnostics/tests/test_worker_plugin.py

```diff
@@ -53,6 +53,32 @@ async def test_create_with_client(c, s):
     assert worker._my_plugin_status == "teardown"
 
 
+@gen_cluster(client=True, nthreads=[])
+async def test_remove_with_client(c, s):
+    await c.register_worker_plugin(MyPlugin(123), name="foo")
+
+    worker = await Worker(s.address, loop=s.loop)
+    assert worker._my_plugin_status == "setup"
+    assert worker._my_plugin_data == 123
```

Comment:
Thinking about it a bit more, I think we can remove these `assert`s since they are testing that the plugin's `setup` method ran which is not the point of this test and is being tested elsewhere (in `test_create_with_client`)

Reply:
Thanks for the comments James. 
Sounds good, in that case should we re-factor the test to include the case of having two plugins and remove one, or do we want to have it as a separate one?

Reply:
Yeah, registering two worker plugins and then removing them one at a time would be good to include here. That way we can make sure that the logic on the scheduler and worker which selects which plugin should be removed is correct.

Reply:
Removed the asserts and made the test to create two plugins and delete one at the time with the proper asserts in between.